include/fmt/std.h

```diff
@@ -389,6 +391,52 @@ struct formatter<
 #endif
   }
 };
+
+namespace detail {
+
+template <typename T, typename Enable = void>
+struct has_flip : std::false_type {};
+
+template <typename T>
+struct has_flip<T, void_t<decltype(std::declval<T>().flip())>>
+    : std::true_type {};
+
+template <typename T> struct is_bit_reference_like {
+  static constexpr const bool value =
+      std::is_convertible<T, bool>::value &&
+      std::is_nothrow_assignable<T, bool>::value && has_flip<T>::value;
+};
+
+}  // namespace detail
+
+FMT_EXPORT
+template <typename BitRef, typename Char>
+struct formatter<BitRef, Char,
+                 enable_if_t<detail::is_bit_reference_like<BitRef>::value>>
+    : formatter<bool, Char> {
+  template <typename FormatContext>
+  FMT_CONSTEXPR auto format(const BitRef& v, FormatContext& ctx) const
+      -> decltype(ctx.out()) {
+    return formatter<bool, Char>::format(static_cast<bool>(v), ctx);
+  }
+};
+
+#ifdef _LIBCPP_VERSION
+
+// Workaround for libc++ incompatibility with C++ standard.
+// According to the Standard, `bitset::operator[] const` returns bool.
+FMT_EXPORT
+template <typename C, typename Char>
+struct formatter<std::__bit_const_reference<C>, Char> : formatter<bool, Char> {
```

Comment:
Similarly, this should be `bitset::reference` and remove ifdef and a comment.

Reply:
See: https://github.com/fmtlib/fmt/pull/3570#discussion_r1285057473

Reply:
I suggest moving this to a separate PR and adding more context why this is needed.

Reply:
> I suggest moving this to a separate PR and adding more context why this is needed.

I do not understand why.
Without this workaround we can't format values from ``const`` containers on libc++.

Reply:
It would make sense to move if we used concrete types but in view of the above discussion it seems impossible so ignore this comment.